plugins/gui/src/GUIWindows.cpp

```diff
@@ -996,24 +996,30 @@ bool megamol::gui::GUIWindows::SynchronizeGraphs(megamol::core::MegaMolGraph* me
                 __LINE__);
         }
 
+        // Check for script path name and for requested update of GUI state
         GraphPtr_t graph_ptr;
         if (graph_sync_success && this->configurator.GetGraphCollection().GetGraph(this->state.graph_uid, graph_ptr)) {
             std::string script_filename = this->state.last_script_filename;
-            if (graph_ptr->GetFilename().empty()) {
-                if (this->core_instance != nullptr) {
-                    // Set project filename from lua state of core instance
-                    script_filename = this->core_instance->GetLuaState()->GetScriptPath();
-                }
+            // Set project filename from lua state of core instance
+            if (graph_ptr->GetFilename().empty() && (this->core_instance != nullptr)) {
+                script_filename = this->core_instance->GetLuaState()->GetScriptPath();
             }
-            // Always check for changed script path when project file is dropped
+            // Always check for changed script path when new project file might be loaded via LuaServiceWrapper
             if (!this->state.project_script_paths.empty()) {
                 script_filename = this->state.project_script_paths.front();
             }
             // Load GUI state from project file when project file changed
             if (script_filename != this->state.last_script_filename) {
                 graph_ptr->SetFilename(script_filename);
-                this->load_state_from_file(graph_ptr->GetFilename());
                 this->state.last_script_filename = script_filename;
+
+                if (megamol::core::utility::FileUtils::FileHasExtension(script_filename, ".png")) {
```

Comment:
We now have 3 different places in three different services where this code is used (copy-pasted). Need to centralize project loading ASAP. Another service?